arangod/Aql/ExecutionEngine.cpp

```diff
@@ -553,10 +550,10 @@ ExecutionEngine* ExecutionEngine::instantiateFromPlan(
       }
     } else {
       // instantiate the engine on a local server
-      engine = new ExecutionEngine(query);
-      auto inst = std::make_unique<Instanciator>(engine);
-      plan->root()->walk(inst.get());
-      root = inst.get()->root;
+      ExecutionEngine engine(query);
```

Comment:
Doesn't this change the behavior?
Now there is a block-local `engine` variable, but the `engine` variable in line 495 is not populated anymore. Previously the `engine` object created here was assigned to the variable declared in line 495.
I think this is an unintended change and it may have side effects.

Reply:
Good catch, I completely missed that! Will fix this.